recipes/elfutils/all/conanfile.py

```diff
@@ -0,0 +1,161 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+import os
+import glob
+
+
+class ElfutilsConan(ConanFile):
+    name = "elfutils"
+    description = "A dwarf, dwfl and dwelf functions to read DWARF, find separate debuginfo, symbols and inspect process state."
+    homepage = "https://sourceware.org/elfutils"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("conan", "elfutils", "libelf", "libdw", "libasm")
+    exports = ["patches/elfutils-*.patch"]
+    license = ["GPL-1.0-or-later", "LGPL-3.0-or-later", "GPL-3.0-or-later"]
+    
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = {
+        "shared": True,
+        "fPIC": True
+    }
+    requires = (
+        "bzip2/1.0.6",
+        "zlib/1.2.11",
+        "xz_utils/5.2.4"
+    )
+
+    _autotools = None
+    _source_subfolder = "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "elfutils" + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_autotools(self):
+        if not self._autotools:
+            args = ['--enable-deterministic-archives', '--enable-silent-rules', '--with-zlib', '--with-bzlib', '--with-lzma']
+            self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
```

Comment:
Something like this?
```
    def build_requirements(self):
        if tools.os_info.is_windows and not os.environ.get("CONAN_BASH_PATH") and \
                tools.os_info.detect_windows_subsystem() != "msys2":
            self.build_requires("msys2/20190524")
```
Plus `win_bash=tools.os_info.is_windows` I guess

Reply:
Indeed.
If the make files assume a gcc compiler, then the MSVC cl compiler needs to be wrapped by some script. (grep for `automake` in the repo)
It is possible that elfutils does not support MSVC at all and you will have to raise a `ConanInvalidConfiguration` when `compiler == "Visual Studio"`. e.g. when it is using `unistd.h` and no gnulib.

Reply:
I just checked your recipe on my Windows box and it appears elfutils requires nested functions,
which is apparently not supported by Visual Studio.

Reply:
Thank you very much for checking! Il will just raise a `ConanInvalidConfiguration` then. What should happen with other compilers though? For instance one of the executions on a mac fails because a gcc check (`configure: error: gcc with GNU99 support required`). Should then all compilers other than gcc just raise a ConanInvalidConfiguration?

Reply:
On Macos, apple-clang is used, which is a fork of clang.
I just tried this recipe on my Linux box, with clang 8.
So I think you you should also raise a ConanInvalidConfiguration when compiler == ["clang", "apple-clang"]
So that sets the number of failing compilers to 3 :smile: